cmd/thanos/bucket.go

```diff
@@ -42,34 +41,28 @@ var (
 func registerBucket(m map[string]setupFunc, app *kingpin.Application, name string) {
 	cmd := app.Command(name, "inspect metric data in an object storage bucket")
 
-	gcsBucket := cmd.Flag("gcs-bucket", "Google Cloud Storage bucket name for stored blocks.").
-		PlaceHolder("<bucket>").String()
-
-	s3Config := s3.RegisterS3Params(cmd)
+	bucketConf := cmd.Flag("objstore.config", "The yaml format configuration of bucket for stored blocks.").
+		PlaceHolder("<bucket.config>").String()
 
 	// Verify command.
 	verify := cmd.Command("verify", "verify all blocks in the bucket against specified issues")
 	verifyRepair := verify.Flag("repair", "attempt to repair blocks for which issues were detected").
 		Short('r').Default("false").Bool()
 	// NOTE(bplotka): Currently we support backup buckets only in the same project.
-	verifyBackupGCSBucket := cmd.Flag("gcs-backup-bucket", "Google Cloud Storage bucket name to backup blocks on repair operations.").
-		PlaceHolder("<bucket>").String()
-	verifyBackupS3Bucket := cmd.Flag("s3-backup-bucket", "S3 bucket name to backup blocks on repair operations.").
-		PlaceHolder("<bucket>").String()
+	backupBucketConf := verify.Flag("objstore-backup.config", "The yaml format configuration of backup bucket for stored blocks.").
+		PlaceHolder("<bucket-backup.config>").String()
 	verifyIssues := verify.Flag("issues", fmt.Sprintf("Issues to verify (and optionally repair). Possible values: %v", allIssues())).
 		Short('i').Default(verifier.IndexIssueID, verifier.OverlappedBlocksIssueID).Strings()
 	verifyIDWhitelist := verify.Flag("id-whitelist", "Block IDs to verify (and optionally repair) only. "+
 		"If none is specified, all blocks will be verified. Repeated field").Strings()
 	m[name+" verify"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
-		bkt, err := client.NewBucket(logger, gcsBucket, *s3Config, reg, name)
+		bkt, err := client.NewBucket(logger, *bucketConf, reg, name)
 		if err != nil {
 			return err
 		}
 		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
 
-		backupS3Config := *s3Config
-		backupS3Config.Bucket = *verifyBackupS3Bucket
```

Comment:
I don't agree. We just want to get client of `bucket`, if not, it will be throw an error. The provider should verify the configuration itself, not in `client` package.